paddle/phi/kernels/kps/logical_kernel.cu

```diff
@@ -25,24 +25,45 @@
 
 namespace phi {
 
-#define DEFINE_LOGICAL_BINARY_KERNEL(type)                            \
-  template <typename T, typename Context>                             \
-  void Logical##type##Kernel(const Context& dev_ctx,                  \
-                             const DenseTensor& x,                    \
-                             const DenseTensor& y,                    \
-                             DenseTensor* out) {                      \
-    if (!out->IsSharedWith(x)) {                                      \
-      dev_ctx.template Alloc<bool>(out);                              \
-    }                                                                 \
-                                                                      \
-    funcs::Logical##type##Functor<T> binary_func;                     \
-    std::vector<const DenseTensor*> ins = {&x, &y};                   \
-    std::vector<DenseTensor*> outs = {out};                           \
-    if (!out->IsSharedWith(x)) {                                      \
-      funcs::BroadcastKernel<bool>(dev_ctx, ins, &outs, binary_func); \
-    } else {                                                          \
-      funcs::BroadcastKernel<T>(dev_ctx, ins, &outs, binary_func);    \
-    }                                                                 \
+template <typename T, typename Context, typename Functor>
+void LogicalKnernelImpl(const Context& dev_ctx,
+                        const DenseTensor& x,
+                        const DenseTensor& y,
+                        DenseTensor* out) {
+  dev_ctx.template Alloc<bool>(out);
+  Functor binary_func;
+  std::vector<const DenseTensor*> ins = {&x, &y};
+  std::vector<DenseTensor*> outs = {out};
+  funcs::BroadcastKernel<bool>(dev_ctx, ins, &outs, binary_func);
+}
+
+template <typename T, typename Context, typename Functor>
+void InplaceLogicalKnernelImpl(const Context& dev_ctx,
+                               const DenseTensor& x,
+                               const DenseTensor& y,
+                               DenseTensor* out) {
```

Comment:
why create `InplaceLogicalKnernelImpl` here while not create in `logical_kernel.cc`, can we integrate code of `LogicalKnernelImpl` and `InplaceLogicalKnernelImpl` into `Logical##type##Kernel` to maintain the same form as `logical_kernel.cc`